frontend/packages/dev-console/src/components/helm/HelmReleaseUpgrade.tsx

```diff
@@ -0,0 +1,179 @@
+export type HelmReleaseUpgradeFormData = {
+  helmReleaseName: string;
+  chartVersion: string;
+  chartValuesYAML: string;
+};
+
+export const HelmReleaseUpgrade: React.FC<HelmReleaseUpgradeProps> = (props) => {
+  const {
+    match: {
+      params: { ns: namespace, releaseName },
+    },
+  } = props;
+  const [chartValues, setChartValues] = React.useState({});
+  const [chartName, setChartName] = React.useState('');
+  const [helmChartData, setHelmChartData] = React.useState<HelmChartData[]>([]);
+  const [chartValuesLoaded, setChartValuesLoaded] = React.useState(false);
+  const [chartVersions, setChartVersions] = React.useState({});
+  const [activeChartVersion, setActiveChartVersion] = React.useState<string>('');
+
+  React.useEffect(() => {
+    let ignore = false;
+
+    const fetchHelmChartValues = async () => {
+      let res: HelmRelease;
+      try {
+        res = await coFetchJSON(`/api/helm/release?ns=${namespace}&release_name=${releaseName}`);
+      } catch {
+        if (ignore) return;
+      }
+      if (ignore) return;
+
+      setActiveChartVersion(res.chart.metadata.version);
+      setChartName(res.chart.metadata.name);
+      setChartValues(res.chart.values);
+      setChartValuesLoaded(true);
+    };
+
+    fetchHelmChartValues();
+
+    return () => {
+      ignore = true;
+    };
+  }, [namespace, releaseName]);
+
+  React.useEffect(() => {
+    let ignore = false;
+
+    const fetchHelmChartVersions = async () => {
+      let json: HelmRelease;
+      try {
+        const res = await coFetch('/api/helm/charts/index.yaml');
+        const yaml = await res.text();
+        json = safeLoad(yaml);
+      } catch {
+        if (ignore) return;
+      }
+      if (ignore) return;
+
+      const versions = _.reduce(
+        _.get(json, ['entries', chartName]),
+        (obj, chart) => {
+          obj[chart.version] = chart.version;
+          return obj;
+        },
+        {},
+      );
+      setHelmChartData(_.get(json, ['entries', chartName]));
+      setChartVersions(versions);
+    };
+    fetchHelmChartVersions();
+
+    return () => {
+      ignore = true;
+    };
+  }, [chartName]);
+
+  const initialValues: HelmReleaseUpgradeFormData = {
+    helmReleaseName: releaseName,
+    chartVersion: activeChartVersion,
+    chartValuesYAML: chartValues ? safeDump(chartValues) : undefined,
+  };
+  const validationSchema = yup.object().shape({
+    chartVersion: yup.string().required('Required'),
+  });
+
+  const handleSubmit = (values, actions) => {
+    const { helmReleaseName, chartVersion, chartValuesYAML }: HelmReleaseUpgradeFormData = values;
+    let valuesObj;
+
+    if (chartValuesYAML) {
+      try {
+        valuesObj = safeLoad(chartValuesYAML);
+      } catch (err) {
+        actions.setStatus({ submitError: `Invalid YAML - ${err}` });
+        return;
+      }
+    }
+    const chartURL: string = getChartURL(helmChartData, chartVersion);
```

Comment:
The upgrade will fail for any chart that was installed via CLI or some other catalog since you assume that you'll find the `chartURL` in one of the charts available in the catalog.